### ISSUE TITLE
Will this work as a drop in replacement for the filesystem in other node modules as well?

### ISSUE THREAD

**Issue by megamindbrian** [jvilk/BrowserFS#181]
This is what I am trying to accomplish:

https://github.com/webpack/memory-fs/issues/43

**Comment by jvilk** [comment]
Not sure I grok 100% what you are trying to accomplish, but this should work as a polyfill for Node's file system module with the following caveats:

* Synchronous API calls are not available on some file systems. You can make any asynchronous file system synchronous with `AsyncMirrorFS`, with the caveat that it needs to asynchronously preload all of the files in the filesystem into a synchronous file system before you can use it (typically the `InMemory` file system).
* Read/write streams are currently not supported.
* Symlinks are not supported on most file systems, although that could change.
* User/group permissions are not supported on many file systems.

Does that answer your question?

**Comment by jvilk** [comment]
Closing for now since I think that's what you were asking. [There's a summary of all available file system backends in the documentation](https://jvilk.com/browserfs/1.3.0/index.html). Some are data stores (`localStorage`, `IndexedDB`, `Dropbox`, etc), while others are meta (`MountableFileSystem` lets you mount file systems into a hierarchy, `AsyncMirrorFS` is what I described in the previous comment, etc).
